### ISSUE TITLE
MTG adapter iOS crashes

### ISSUE THREAD

**Issue by nechtas** [cleveradssolutions/CAS-Flutter#6]
See a significant number of crashes on the iOS platform with such stack trace:

```
Fatal Exception: NSInternalInconsistencyException
0  CoreFoundation                 0xed5e0 __exceptionPreprocess
1  libobjc.A.dylib                0x2bc00 objc_exception_throw
2  CoreAutoLayout                 0x57ac NSISSparseVectorAddTermWithPlaceValueCoefficientStartingIndex
3  CoreAutoLayout                 0x11e84 -[NSISEngine removeConstraintWithMarker:]
4  CoreAutoLayout                 0x11df8 -[NSLayoutConstraint _removeFromEngine:]
5  UIKitCore                      0x1e20e4 -[UIView _layoutEngine_willRemoveLayoutConstraint:]
6  UIKitCore                      0x4ed84 -[UIView _setAutoresizingConstraints:]
7  UIKitCore                      0xa008c -[UIView _invalidateAutoresizingConstraints]
8  UIKitCore                      0x820b8 -[UIView(UIConstraintBasedLayout) _constraints_subviewWillChangeSuperview:]
9  UIKitCore                      0x11c1b08 -[UIWindow(AdditionalLayoutSupport) _constraints_subviewWillChangeSuperview:]
10 UIKitCore                      0x7f660 __UIViewWillBeRemovedFromSuperview
11 UIKitCore                      0x7f16c -[UIView(Hierarchy) removeFromSuperview]
12 Runner                         0x1661ce8 -[MTGWebViewCacheManager removeWebViewWithUrl:]
13 Runner                         0x15b1eb4 +[MTGVideoAdResourcePreparer removeWebviewCacheWithPathURL:]
14 Runner                         0x15b1f54 +[MTGVideoAdResourcePreparer removeWebviewCacheWithCampaign:unitID:]
15 Runner                         0x15a5c40 -[MTGVideoAdRequestComposite releasePreviousResource:newCampaigns:]
16 Runner                         0x15a590c -[MTGVideoAdRequestComposite releasePreviousResourceAndSaveCurrentResource:]
17 Runner                         0x15a8dc4 -[MTGVideoAdRequestComposite hitLocalBackupCacheLoginWithLoadFailedRequestId:loadFailedLocalRequestId:scenes:]
18 Runner                         0x15a3000 -[MTGVideoAdRequestComposite isHitPriorityToCallbacks]
19 Runner                         0x15a3d00 -[MTGVideoAdRequestComposite requestApiAdWithTimeout:]
20 Runner                         0x16c3854 -[MTGAdRequestDelayManager performDelayExcuteBlock:]
21 Runner                         0x159f6d8 -[MTGVideoAdManagerComposite _loadVideoWithBidToken:placementId:unitId:delegate:autoLoad:]
22 Runner                         0x170252c -[MTGBidInterstitialVideoAdManager loadAdWithBidToken:]
23 Runner                         0x17035e4 -[MTGNewInterstitialBidAdManager loadAdWithBidToken:]
24 Runner                         0x1584bc CASMintegralInterstitial.requestAd() + 32 (CASMintegralInterstitial.swift:32)
25 Runner                         0x6d82d4 MediationAgent.beginRequest() + 220 (MediationAgent.swift:220)
26 Runner                         0x6e07f0 AgentTimeout.beginRequest(_:timeDivider:) + 17 (AgentTimeout.swift:17)
27 Runner                         0x6cf3dc BiddingHandler.loadAdContent(_:) + 236 (BiddingHandler.swift:236)
28 Runner                         0x72d1bc AuctionNoticeHandelr.response(_:) + 32 (AuctionNoticeHandler.swift:32)
29 Runner                         0x157b50 CASMintegralBidUnit.sendNotice(_:) + 99 (CASMintegralBidUnit.swift:99)
30 Runner                         0x6cec94 BiddingHandler.continueWithWon(_:_:) + 22 (AuctionNoticeHandler.swift:22)
31 Runner                         0x6e7bec MediationController.trySelectAdWinner() + 309 (MediationController.swift:309)
32 Runner                         0x7041b0 WaterfallHandler.onWaterfallDone(_:) + 259 (WaterfallHandler.swift:259)
33 Runner                         0x7044d0 WaterfallHandler.this() + 286 (WaterfallHandler.swift:286)
34 Runner                         0x704acc WaterfallHandler.onFailedToLoad(_:) + 326 (WaterfallHandler.swift:326)
35 Runner                         0x6d89ec MediationAgent.onRequestFailed(_:_:_:) + 272 (MediationAgent.swift:272)
36 Runner                         0x6d9e48 closure #1 in MediationAgent.onAdFailedToLoad(_:_:_:)
37 Runner                         0x12daf4 thunk for @escaping @callee_guaranteed () -> () (<compiler-generated>)
38 libdispatch.dylib              0x26a8 _dispatch_call_block_and_release
39 libdispatch.dylib              0x4300 _dispatch_client_callout
40 libdispatch.dylib              0xb894 _dispatch_lane_serial_drain
41 libdispatch.dylib              0xc3c4 _dispatch_lane_invoke
42 libdispatch.dylib              0x17004 _dispatch_root_queue_drain_deferred_wlh
43 libdispatch.dylib              0x16878 _dispatch_workloop_worker_thread
44 libsystem_pthread.dylib        0x1964 _pthread_wqthread
45 libsystem_pthread.dylib        0x1a04 start_wqthread
```

**Comment by Str4tos** [comment]
Hello @nechtas, thank you for creating the issue.  
It seems to be an issue within the Mintegral Framework. We need more information about the problem to escalate it for Mintegral to address.

1. What is the CAS version?
2. Which iOS versions are affected by this issue?

**Comment by nechtas** [comment]
Hi @Str4tos.
Flutter package version is 0.2.3, ios pod version is 'CleverAdsSolutions-SDK/Optimal', '~> 3.5.1'.
It's mostly iOS 17, and also small portion of iOS 16.

**Comment by Str4tos** [comment]
Thank you, we will fix this issue in the next update CAS 3.5.2.
